modules/ding_wayf_dk/modules/ding_wayf_dk_create_user/ding_wayf_dk_create_user.module

```diff
@@ -0,0 +1,397 @@
+        'link' => NULL,
+      ),
+      'path' => $path . '/templates',
+    ),
+  );
+}
+
+/**
+ * Menu callback for information page before user creation.
+ */
+function ding_wayf_dk_create_user_info_page() {
+  // Get texts from the configuration.
+  $values = variable_get('ding_wayf_user_create_before_info', array());
+
+  return array(
+    '#theme' => 'ding_wayf_dk_info_page',
+    '#title' => empty($values['title']) ? t('Not configured.') : $values['title'],
+    '#content' => empty($values['content']) ? t('Please configure WAYF configuration to get content on this page under <a href="/admin/config/people/wayf">admin/config/people/wayf</a>.') : $values['content'],
+    '#link' => array(
+      '#theme' => 'link',
+      '#text' => empty($values['link']) ? t('Register as new user') : $values['link'],
+      '#path' => 'wayf/consume',
+      '#options' => array(
+        'attributes' => array(
+          'class' => array(),
+        ),
+        'query' => array(
+          'destination' => 'wayf/create-user/form',
+        ),
+        'html' => TRUE,
+      ),
+    ),
+  );
+}
+
+/**
+ * Menu callback for information page after user creation.
+ */
+function ding_wayf_dk_create_user_notice() {
+  // Get texts from the configuration.
+  $values = variable_get('ding_wayf_user_create_after_info', array());
+
+  return array(
+    '#theme' => 'ding_wayf_dk_info_page',
+    '#title' => empty($values['title']) ? t('Not configured.') : $values['title'],
+    '#content' => empty($values['content']) ? t('Please configure WAYF configuration to get content on this page under <a href="/admin/config/people/wayf">admin/config/people/wayf</a>.') : $values['content'],
+  );
+}
+
+/**
+ * Menu callback to create the user creation form.
+ */
+function ding_wayf_dk_create_user_form() {
+  // Remove all message from the normal login, that have not succeed.
+  drupal_get_messages(NULL, TRUE);
+
+  // Check if user have been logged in.
+  if (user_is_logged_in()) {
+    // User is logged in and have an account.
+    drupal_set_message(t('You already have an user account at the library. So we have just logged you in.'));
+
+    // Ensure session information is removed.
+    unset($_SESSION['ding_wayf_dk_create_user']);
+    unset($_SESSION['ding_wayf_dk_login_destination']);
+
+    // Send user to front page.
+    drupal_goto('<front>');
+  }
+
+  // This is a little hack to get around the panels design requirements of the
+  // ddbasic theme.
+  $form = array(
+    '#prefix' => '<div class="default-layout"><div class="layout-wrapper">',
+    '#suffix' => '<br /></div></div>',
+  );
+
+  $form['accept'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('I accept the user agreements'),
+    '#description' => variable_get('ding_wayf_user_create_accpect_text', ''),
+    '#required' => TRUE,
+  );
+
+  $form['name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Name'),
+    '#description' => t('You full name.'),
+    '#maxlength' => 100,
```

Comment:
Why this `maxlength`? If it is caused by other Drupal limitations then please add a comment stating why.